modules/common/src/test/scala/com/snowplowanalytics/snowplow/storage/bigquery/common/CachedFieldSpec.scala

```diff
@@ -0,0 +1,201 @@
+
+    "(3) schema is patched, delay between getSchemaAsField calls is less than cache TTL => original schema is still cached => 1 field cached" in {
+      val fieldCache = getCache
+      val resolver = getResolver
+
+      //first call
+      getSchemaAsField(fieldCache, resolver)(
+        currentTime = 1000, //ms
+        schemaInRegistry = `original schema - 1 field`
+      )
+
+      //second call, 2s later, less than 10s TTL
+      val result = getSchemaAsField(fieldCache, resolver)(
+        currentTime = 3000, //ms
+        schemaInRegistry = `patched schema - 2 fields` //different schema with the same key!
+      )
+
+      //no data from patched schema
+      result must beEqualTo(fieldFor(`original schema - 1 field`))
+
+      //Properties are cached after first call (1 second)
+      fieldCache.get((schemaKey, 1.second)) must beSome
+
+      //Properties are not cached after second call (3 seconds)
+      fieldCache.get((schemaKey, 3.second)) must beNone
+    }
+
+    "(4) schema is patched, delay between flatten calls is greater than cache TTL => original schema is expired => using patched schema => 2 fields cached" in {
+      val fieldCache = getCache
+      val resolver = getResolver
+
+      //first call
+      getSchemaAsField(fieldCache, resolver)(
+        currentTime = 1000, //ms
+        schemaInRegistry = `original schema - 1 field`
+      )
+
+      //second call, 12s later - greater than 10s TTL
+      val result = getSchemaAsField(fieldCache, resolver)(
+        currentTime = 13000, //ms
+        schemaInRegistry = `patched schema - 2 fields` //different schema with the same key!
+      )
+
+      //Cache content expired, patched schema is fetched => expected field is based on the patched schema
+      result must beEqualTo(fieldFor(`patched schema - 2 fields`))
+
+      //Properties are cached after first call (1 second)
+      fieldCache.get((schemaKey, 1.second)) must beSome
+
+      //Properties are cached after second call (13 seconds)
+      fieldCache.get((schemaKey, 13.second)) must beSome
+    }
+  }
+
+  private def fieldFor(json: Json): Field = DdlSchema.parse(json).map(schema => Field.build("", schema, false)).get
+  //Helper method to wire all test dependencies and execute LoaderRow.getSchemaAsField
+  private def getSchemaAsField(fieldCache: FieldCache[Id], resolver: Resolver[Id])(schemaInRegistry: Json, currentTime: Long): Field = {
+
+    //To return value stored in the schemaInRegistry variable, passed registry is ignored
+    val testRegistryLookup: RegistryLookup[Id] = new RegistryLookup[Id] {
+      override def lookup(registry: Registry, schemaKey: SchemaKey): Id[Either[RegistryError, Json]] = Right(schemaInRegistry)
+
+      override def list(registry: Registry,
+                        vendor: String,
+                        name: String,
+                        model: Int): Id[Either[RegistryError, SchemaList]] = Left(NotFound) // not used
+    }
+
+    val idClock: Clock[Id] = new Clock[Id] {
```

Comment:
I think since usually we need a clock that actually knows the current time, and sometimes a clock that always thinks it is the same time we set (as we do in this test) I've moved the clock over to `SpecHelpers` as `stoppedTimeClock`.